### PR TITLE
Added Spawning example

### PR DESCRIPTION
**Pull request by Nereuxofficial**
I added a chapter on spawning. Let me know if there is anything i should add

### CODE REVIEW

File: src/06_multiple_futures/04_spawning.md
Comment (mfloto) [inline src/06_multiple_futures/04_spawning.md:15]: This sentence seems incomplete. Could you clarify what is provided?

**Review by mfloto — CHANGES_REQUESTED** [review]

File: src/06_multiple_futures/04_spawning.md
Comment (Nereuxofficial) [reply src/06_multiple_futures/04_spawning.md:15]: I i originally wanted to add a link, but deleted the entire word. Thanks for pointing this out

**Review by mfloto — APPROVED** [review]

**Review by Flippchen — APPROVED** [review]
LGTM

**Comment by Nereuxofficial** [comment]
@eholk Could i get some feedback on this?

File: ci/dictionary.txt
Comment (eholk) [inline ci/dictionary.txt:106]: It looks like this file is in alphabetical order. Would you mind moving JoinHandle up to about line 40?

File: src/06_multiple_futures/04_spawning.md
Comment (eholk) [inline src/06_multiple_futures/04_spawning.md:9]: It might be worth discussing some of the task life cycle questions around spawn. For example, what happens if main finishes before a the spawned task does in the example? Does the whole program block until the task is finished, or do the spawned tasks get cancelled?

File: src/06_multiple_futures/04_spawning.md
Comment (eholk) [inline src/06_multiple_futures/04_spawning.md:16]: It'd probably be good to have an example showing this, or linking to an existing one if there's already one in the book.

**Review by eholk — COMMENTED** [review]
Hey, thanks for the contribution! Overall I think this is a nice example. I made a few suggestions for how to improve it, which you're welcome to do or feel free to say you'd like to hold off on those for a later PR (either by you or another author).

File: src/06_multiple_futures/04_spawning.md
Comment (Nereuxofficial) [reply src/06_multiple_futures/04_spawning.md:16]: I thought about that and linking the async_std or tokio doc about channels but since the link may change it would be nice to have a reference to another book chapter, however i could absolutely make a short example for it in this chapter

File: src/06_multiple_futures/04_spawning.md
Comment (Nereuxofficial) [reply src/06_multiple_futures/04_spawning.md:9]: That section is now added with an example. Thanks for the feedback!

**Comment by Nereuxofficial** [comment]
Thanks for the feedback. I implemented most of the changes suggested although for the channels a chapter on them would probably be nice as outside links can change pretty fast without the book getting notice, whereas if a chapter name changed it would be clear that the links to it need to change too

**Comment by Nereuxofficial** [comment]
@eholk could i get another review on this? I added a new section

**Comment by eholk** [comment]
Hey @Nereuxofficial - thanks for the extra section. I don't think I'll be able to review it this week but I'll try to get to it next week.

**Comment by blandger** [comment]
Whats stops from merging new section ? A month has passed....

**Comment by eholk** [comment]
Sorry about my slow review. This looks good! I'll go ahead and merge it.

**Pull request merged**
